routes/index.js

```diff
@@ -97,7 +98,48 @@ module.exports = function (passport) {
   /* POST registration page. Uses form data to create a new user in the database.
    *      temporarily outputs the results of dbAPI method to console until alerts can be added.
    * ********************************************************************************************/
-  router.post('/registration/:badge', jsonParser, function (req, res) {
+  router.post('/registration/:badge', [
+    check('badge')
+      .exists()
+      .custom(value => {
+        return dbAPI.validateUser(value).then(user => {
+          if (user != undefined)
+            throw new Error('The badge you provided is already in use');
+        });
+      }),
+
+    check('first').exists().withMessage('Please enter first name'),
+
+    check('last').exists().withMessage('Please enter last name'),
+
+    check('email')
+      .exists()
+      .isEmail().withMessage('Please enter valid email')
+      .trim()
+      .normalizeEmail()
+      .custom(value => {
+        return dbAPI.validateUser(value).then(user => {
+          if (user != undefined)
+            throw new Error('The email you provided is already in use');
+        });
+      }),
+
+    check('phone')
+      .exists()
+      .isMobilePhone('any').withMessage('Please enter valid phone number'),
+
+    check('signature').exists().withMessage('Please enter signature'),
+
+    check('ecSignature').exists(),
```

Comment:
exists() just checks whether or not there is a field coming in the form data that is called ecSignature. exists() doesnt care whether or not that field is empty. I dont think it hurts for now. You can still leave it black on the registration and it will accept your form.